roamance/src/context/SocialContext.tsx

```
@@ -0,0 +1,343 @@
+          // Optimistically update like state
+          setPosts((prev) =>
+            prev.map((p) =>
+              p.id === postId
+                ? {
+                    ...p,
+                    likes_count: isLiked
+                      ? Math.max((p.likes_count || 0) - 1, 0)
+                      : (p.likes_count || 0) + 1,
+                    liked_by: isLiked
+                      ? p.liked_by.filter((u) => u.id !== user.id)
+                      : [...p.liked_by, user!],
```

Biome (1.9.4)

[error] Forbidden non-null assertion.



(lint/style/noNonNullAssertion)